creating_images/custom.adoc

```diff
@@ -0,0 +1,40 @@
+= Custom builder
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+link:../architecture/core_objects/builds.html#custom-build[Custom build] is designed
+to fill the gap that was created when everybody jumped into creating docker images.
+Still there is a requirement to produce packages with all the software being installed
+and most importantly one needs to be able to build base docker images. This is
+where Custom build is the perfect match to fill in that gap.
```

Comment:
I'll update this in a follow up PR.
On Jun 15, 2015 6:58 PM, "Ben Parees" <notifications@github.com> wrote:

> In creating_images/custom.adoc
> <https://github.com/openshift/openshift-docs/pull/492#discussion_r32441856>
> :
>
> > +{product-author}
> > +{product-version}
> > +:data-uri:
> > +:icons:
> > +:experimental:
> > +:toc: macro
> > +:toc-title:
> > +
> > +toc::[]
> > +
> > +== Overview
> > +link:../architecture/core_objects/builds.html#custom-build[Custom build] is designed
> > +to fill the gap that was created when everybody jumped into creating docker images.
> > +Still there is a requirement to produce packages with all the software being installed
> > +and most importantly one needs to be able to build base docker images. This is
> > +where Custom build is the perfect match to fill in that gap.
>
> this isn't really how i'd describe custom builder... the role it fills is
> basically anything we may not have anticipated in our STI and Docker
> builders. In addition the most likely use case is not to produce a new
> docker image (use Docker or STI builds for that), it's to produce
> individual artifacts like jars, wars, installable zips, etc. The point is
> that it can perform any steps you want to produce some output artfact, and
> then is responsible for pushing that artifact to some location (eg scp'ing
> it to a server).
>
> It also provides a backdoor way of introducing a CI/CD flow since you
> could write a custom builder that runs unit or integration tests, in
> theory. (sort of like the Jenkins custom job type)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/492/files#r32441856>.
>